src/InstrumentationEngine.Attach/EngineConfiguration.xsd

```diff
@@ -0,0 +1,47 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified" xmlns:xs="http://www.w3.org/2001/XMLSchema">
+    <!-- root type -->
+    <xs:element name="InstrumentationEngineConfiguration">
+        <xs:complexType>
+            <xs:sequence>
+                <xs:element name="InstrumentationEngine">
+                    <xs:complexType>
+                        <xs:sequence>
+                            <xs:element name="Settings" type="settingsType" />
+                        </xs:sequence>
+                    </xs:complexType>
+                </xs:element>
+                <xs:element name="InstrumentationMethods" type="instrumentationMethodsType" />
+            </xs:sequence>
+        </xs:complexType>
+    </xs:element>
+
+    <!-- level 1 types -->
+    <xs:complexType name="instrumentationMethodsType">
+        <xs:sequence>
+            <xs:element name="AddInstrumentationMethod" minOccurs="0" maxOccurs="unbounded">
+                <xs:complexType>
+                    <xs:sequence>
+                        <xs:element name="Settings" type="settingsType" />
```

Comment:
>Settings [](start = 42, length = 8)

Whoops, the ConfigurationSource schema allows 0 <settings> nodes but if exists it expects at least 1 <setting> node. The EngineConfiguration here has it flipped (requires <Settings> section but allows 0 <setting> nodes).

I'm fixing the EngineConfiguration to allow 0 <settings> as that plays nice with the parsing logic.